### ISSUE TITLE
Avoid storing secret API keys in nix store in oink module

### ISSUE THREAD

**Issue by lunik1** [NixOS/nixpkgs#340376]
## Issue description

There doesn't currently seem to be a way to set a secret API key in the `oink` module without adding it to the nix store, where it will be world-readable. Would it be possible to add an option to set the config file directly, which can then be managed by e.g. `sops.nix`. Alternatively, there could be a `secretApiKeyFile` (and maybe `apiKeyFile`) option which would allow reading those options from a file, but I am not sure how best to pass those on to Oink! Usually, you would add them to the environment of the service, but I am not sure that Oink! supports reading these values from the environment?

Pinging module author @jtbx

**Comment by jtbx** [comment]
Yeah to my knowledge oink only supports passing keys through the config file. It wouldn't be hard to add a config override option but that sort of defeats the purpose of the module.

**Comment by a-kenji** [comment]
@lunik1 
Sops-nix has a templating feature, which allows you to create
the whole file exactly for applications that don't think about security like this. That being said templating the config file just for secrets is not a great user experience and should just be fixed upstream if possible.

**Comment by lunik1** [comment]
Using that would still require the module to expose using a config file directly, though (without restoring to overriding `ExecStart`).

I'll see if there's any appetite for supporting environment variables upstream.

**Comment by RLado** [comment]
Hello, author of Oink here. I can confirm that Oink only supports passing API keys through a configuration file; nevertheless, there is the possibility to override the configuration file using `oink -c <path to config.json>`. 

I am not familiar with Nix but why can't you set the permission of the configuration file to disallow arbitrary reading? I am not against environment variables but I don't understand the safety benefits over the existing implementation.

**Comment by a-kenji** [comment]
> I am not against environment variables but I don't understand the safety benefits over the existing implementation.

The issue got opened here, because this is especially a problem for us.
The nix store is public, if configuration files end up there, they are world readable.

The user experience we want to provide is for people to be able to edit their configuration, without exposing their secret keys. Security and usability are very much linked.

If you allow secrets to be loaded through environment variables your users get the option to benefit from the added security on all distributions - not just NixOS.

There are still more benefits that touch usability and security that I did not explicitly go in here.

Thank you for being curious, reaching out and trying to understand our problems.

**Comment by RLado** [comment]
Thanks for explaining. 

I have added the possibility to override the global API keys using `OINK_OVERRIDE_SECRETAPIKEY` and `OINK_OVERRIDE_APIKEY` environment variables. Hopefully this addresses your use case.

Let me know if you have any other comments or suggestions regarding this change.
